doc/static-route/SONiC_static_route_hdl.md

```diff
@@ -0,0 +1,452 @@
+
+
+A new sonic-static-route.yang is defined to store entries in STATIC_ROUTE table in configDB.
+
+```diff
+
+module: sonic-static-route
+  +--rw sonic-static-route
+     +--rw STATIC_ROUTE
+        +--rw STATIC_ROUTE_LIST* [vrf-name prefix]
+           +--rw vrf-name   union
+           +--rw prefix     inet:ip-prefix
+           +--rw nexthop   string
+           +--rw ifname    string
+           +--rw distance  string
+           +--rw nexthop-vrf string
+           +--rw blackhole   string
+
+
+```
+
+### 3.6.2 CLI
+#### 3.6.2.1 Configuration Commands
+#### 3.6.2.2 ip/ipv6 route config command
+ip route command is used to configure IPv4  static routes in SONiC.
+ipv6 route command is used to configure IPv6 static routes in SONiC.
+##### 3.6.2.2.1
+Syntax
+
+Vrf (for default instance) and distance metric are optional in the CLI.
+
+```
+ip route [vrf <vrf-name>] <prefix: A.B.C.D/mask> {[interface <interface-name>] | [<next-hop-ip>] | [<next-hop-ip> interface <interface-name>] | [blackhole]} [nexthop-vrf <vrf-name>] <distance Metric>
+```
+
+```
+ipv6 route [vrf <vrf-name>] <prefix: A.B.C.D/mask> {[interface <interface-name>] | [<next-hop-ip>] | [<next-hop-ip> interface <interface-name>] | [blackhole]} [nexthop-vrf <vrf-name>] <distance Metric>
+
+```
+
+
+Syntax Description:
+
+|    Keyword    | Description |
+|:-----------------:|:-----------:|
+| vrf | The static routes will be configured in the specified VRF. If not specified, default VRF will be used.
+| prefix | This is the destination prefix for IPv4 or IPv6 routes.
+| interface | Specifies the interface to be used for this route. Specify nexthop-vrf correctly if interface is in another VRF.
+| next-hop-ip | This provides the gateway IP for the prefix. Specify nexthop-vrf correctly if this IP is in another VRF.
+| distance Metric| Specifies distance value for this route. Value from 1 to 255.
+|nexthop-vrf| Specifies nexthop-vrf if interface or next-hop-ip is in another VRF.
+|blackhole| Specifies if this nexthop is to be blackholed. Only distance metric is relevant for blackhole.
+
+
+Example:
+
+#### IPv4 examples
+````
+sonic(config)# ip route 10.1.1.1/24 10.1.1.3
+
+In configDB new STATIC_ROUTE table will be filled with following entries for default VRF and prefix:-
+key                 = STATIC_ROUTE|default|10.1.1.1/24;
+vrf-name            = default
+prefix              = 10.1.1.1/24
+nexthop             = 10.1.1.3
+ifname              = ""
+distance            = 0
+nexthop-vrf         = ""
+blackhole           = false
+
+
+sonic(config)# ip route 10.1.1.1/24 Ethernet 12 nexthop-vrf Vrf-RED  20
+Assumption is Ethernet12 is bound to Vrf-RED.
+STATIC_ROUTE table entries are updated for newly added route:-
+key                 = STATIC_ROUTE|default|10.1.1.1/24;
+vrf-name            = default

```

Comment:
The vrf name and prefix are already in the key. Is it needed to repeat these entries again?

Reply:
No, will remove it.

Reply:
done